Cesium3DTilesContent/src/I3dmToGltfConverter.cpp

```diff
@@ -0,0 +1,777 @@
+        instances[i] = glm::translate(instances[i], transVec);
+      }
+    } else {
+      return errorOut("invalid accessor for instance translations");
+    }
+  }
+  if (rotations) {
+    auto quatAccessorView = getQuaternionAccessorView(model, rotations);
+    std::visit(
+        [&](auto&& arg) {
+          for (unsigned i = 0; i < count; ++i) {
+            auto quat = toGlmQuat<glm::dquat>(arg[i]);
+            instances[i] = instances[i] * glm::toMat4(quat);
+          }
+        },
+        quatAccessorView);
+  }
+  if (scales) {
+    AccessorView<CesiumGltf::AccessorTypes::VEC3<float>> scaleAccessor(
+        model,
+        *scales);
+    if (scaleAccessor.status() == AccessorViewStatus::Valid) {
+      for (unsigned i = 0; i < count; ++i) {
+        auto scaleFactors = toGlm<glm::dvec3>(scaleAccessor[i]);
+        instances[i] = glm::scale(instances[i], scaleFactors);
+      }
+    } else {
+      return errorOut("invalid accessor for instance translations");
+    }
+  }
+  return instances;
+}
+
+const size_t rotOffset = sizeof(glm::vec3);
+const size_t scaleOffset = rotOffset + sizeof(glm::quat);
+const size_t totalStride =
+    sizeof(glm::vec3) + sizeof(glm::quat) + sizeof(glm::vec3);
+
+void copyToBuffer(
+    const glm::dvec3& position,
+    const glm::dquat& rotation,
+    const glm::dvec3& scale,
+    std::byte* bufferLoc) {
+  glm::vec3 fposition(position);
+  std::memcpy(bufferLoc, &fposition, sizeof(fposition));
+  glm::quat frotation(rotation);
+  std::memcpy(bufferLoc + rotOffset, &frotation, sizeof(frotation));
+  glm::vec3 fscale(scale);
+  std::memcpy(bufferLoc + scaleOffset, &fscale, sizeof(fscale));
+}
+
+void copyToBuffer(
+    const glm::dvec3& position,
+    const glm::dquat& rotation,
+    const glm::dvec3& scale,
+    std::vector<std::byte>& bufferData,
+    size_t i) {
+  copyToBuffer(position, rotation, scale, &bufferData[i * totalStride]);
+}
+
+void instantiateGltfInstances(
+    GltfConverterResult& result,
+    const DecodedInstances& decodedInstances) {
+  std::set<CesiumGltf::Node*> meshNodes;
+  int32_t instanceBufferId = createBufferInGltf(*result.model);
+  auto& instanceBuffer =
+      result.model->buffers[static_cast<uint32_t>(instanceBufferId)];
+  int32_t instanceBufferViewId = createBufferViewInGltf(
+      *result.model,
+      instanceBufferId,
+      0,
+      static_cast<int64_t>(totalStride));
+  auto& instanceBufferView =
+      result.model->bufferViews[static_cast<uint32_t>(instanceBufferViewId)];
+  const auto numInstances =
+      static_cast<uint32_t>(decodedInstances.positions.size());
+  auto upToZ = CesiumGltfContent::GltfUtilities::applyGltfUpAxisTransform(
+      *result.model,
+      glm::dmat4x4(1.0));
+  result.model->forEachPrimitiveInScene(
+      -1,
+      [&](Model& gltf,
+          Node& node,
+          Mesh&,
+          MeshPrimitive&,
+          const glm::dmat4& transform) {
+        auto [nodeItr, notSeen] = meshNodes.insert(&node);
+        if (!notSeen) {
```

Comment:
Could this be renamed to a variable without "not" in the name? It is a little confusing to read logic that resolves to "if not not ___"

Maybe something like...
```suggestion
        auto [nodeItr, firstTimeSeen] = meshNodes.insert(&node);
        if (!firstTimeSeen) {
```

Reply:
I changed `notSeen` to `inserted`. This matches the semantics of the result of `insert`, and I think the code reader can understand the intent.